.github/workflows/workflow.yml

```diff
@@ -9,26 +9,69 @@ on:
     paths-ignore:
       - '**.md'
 jobs:
-  run:
-    name: Run
-    runs-on: ${{ matrix.operating-system }}
+  test:
+    name: Unit Tests - ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        operating-system: [ubuntu-latest, windows-latest]
+        os: [ubuntu-latest, macOS-latest, windows-latest]
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v2
 
-    - name: Set Node.js 12
-      uses: actions/setup-node@v1
-      with:
-        node-version: 12
+      - name: Set Node.js 12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      # http://www.tiernok.com/posts/2019/faster-npm-installs-during-ci/
+      - run: npm ci --prefer-offline --no-audit --progress=false
+      - run: npm run pre-push
```

Comment:
I pulled in npm-run-all to have cross platform support for running multiple scripts (linting, type checking, format checking, test-suite, etc). I bundled everything into two scripts, "pre-push" (which runs everything) and "pre-commit" which does the minimal necessary before a commit.

I can expand that back out or rename it if it would be clearer. It was helpful for shaving a few seconds off which improved incremental dev experience, but it's certainly not essential.

---

The current scripts, for reference:

```json
  "scripts": {
    "check:type": "tsc --noEmit",
    "check:format": "prettier --check **/*.{ts,json}",
    "check:lint": "eslint src/**/*.ts",
    "test": "jest",
    "format": "prettier --write **/*.{ts,json}",
    "pack": "ncc build",
    "_git-add": "git add dist",
    "pre-push": "run-p check:* test",
    "pre-commit": "npm-run-all format pack _git-add"
  }
```

Reply:
Ahh, I missed that (and was confused by the name `pre-push`). I'm 👍 to give it a more intentional revealing name or just directly calling `run-p check:* test` from the actions yaml here.

Reply:
I ended up switching to use lint-staged in order to make commit hooks run much more quickly. As a nice side effect, it became very easy to just support `npm test`. Which, after all, is something that will be expected behavior so I'm glad it works "as expected" out of the box now.